src/HonzaBotner.Discord.Services/Commands/MessageCommands.cs

```diff
@@ -92,5 +94,71 @@ public MessageCommands(ILogger<MessageCommands> logger)
         }
+
+        [Group("bind")]
+        [ModuleLifespan(ModuleLifespan.Transient)]
+        public class RoleBindingsCommands : BaseCommandModule
+        {
+            private readonly IRoleBindingsService _roleBindingsService;
+
+            public RoleBindingsCommands(IRoleBindingsService roleBindingsService)
+            {
+                _roleBindingsService = roleBindingsService;
+            }
+
+            [Command("add")]
+            public async Task AddBinding(CommandContext ctx, [Description("URL of the message.")] string url,
+                [Description("Emoji to react with.")] DiscordEmoji emoji,
+                [Description("Roles which will be toggled after reaction")] params DiscordRole[] roles)
+            {
+                DiscordMessage? message = await DiscordHelper.FindMessageFromLink(ctx.Guild, url);
+                if (message == null)
+                {
+                    throw new ArgumentOutOfRangeException($"Couldn't find message with link: {url}");
+                }
+
+                ulong channelId = message.ChannelId;
+                ulong messageId = message.Id;
+
+                await _roleBindingsService.AddBindingsAsync(channelId, messageId, emoji.Name,
+                    roles.Select(r => r.Id).ToHashSet());
+            }
+
+            [Command("remove")]
+            [Priority(1)]
+            public async Task RemoveBinding(CommandContext ctx, [Description("URL of the message.")] string url,
+                [Description("Emoji to react with.")] DiscordEmoji emoji)
+            {
+                DiscordMessage? message = await DiscordHelper.FindMessageFromLink(ctx.Guild, url);
+                if (message == null)
+                {
+                    throw new ArgumentOutOfRangeException($"Couldn't find message with link: {url}");
+                }
+
+                ulong channelId = message.ChannelId;
+                ulong messageId = message.Id;
+
+                await _roleBindingsService.RemoveBindingsAsync(channelId, messageId, emoji.Name, null);
+            }
+
+            [Command("remove")]
+            [Priority(2)]
+            public async Task RemoveBinding(CommandContext ctx, [Description("URL of the message.")] string url,
+                [Description("Emoji to react with.")] DiscordEmoji emoji,
+                [Description("Roles which will be toggled after reaction")] params DiscordRole[] roles)
+            {
+                DiscordMessage? message = await DiscordHelper.FindMessageFromLink(ctx.Guild, url);
+                if (message == null)
+                {
+                    throw new ArgumentOutOfRangeException($"Couldn't find message with link: {url}");
+                }
+
+                ulong channelId = message.ChannelId;
+                ulong messageId = message.Id;
+
+                await _roleBindingsService.RemoveBindingsAsync(channelId, messageId, emoji.Name,
+                    roles.Select(r => r.Id).ToHashSet());
+            }
```

Comment:
Can we merge this? Maybe use `params DiscordRole[] roles = null`?

Reply:
I was thinking about. Need to test it before I will do it. I have done similar thing in the service.

Reply:
Done this here https://github.com/fit-ctu-discord/honza-botner/blob/master/src/HonzaBotner.Discord.Services/Commands/ChannelCommands.cs#L31 :)

Reply:
It works even without the nullable, but we should mark it as nullable I think, so the API process it as optional.

Reply:
#88

Reply:
Not sure what you wanted to say by this